packages/image_picker/image_picker/ios/Classes/FLTPHPickerImageHandlingOperation.m

```diff
@@ -0,0 +1,132 @@
+  self->executing = isExecuting;
+  [self didChangeValueForKey:@"isExecuting"];
+}
+
+- (void)completeOperation {
+  [self setExecuting:NO];
+  [self setFinished:YES];
+}
+
+- (void)start {
+  if ([self isCancelled]) {
+    [self setFinished:YES];
+    return;
+  }
+  [self setExecuting:YES];
+  if (@available(iOS 14, *)) {
+    [self.result.itemProvider
+        loadObjectOfClass:[UIImage class]
+        completionHandler:^(__kindof id<NSItemProviderReading> _Nullable image,
+                            NSError *_Nullable error) {
+          if ([image isKindOfClass:[UIImage class]]) {
+            __block UIImage *localImage = image;
+            PHAsset *originalAsset =
+                [FLTImagePickerPhotoAssetUtil getAssetFromPHPickerResult:self.result];
+
+            if (self.maxWidth != (id)[NSNull null] || self.maxHeight != (id)[NSNull null]) {
+              localImage = [FLTImagePickerImageUtil scaledImage:localImage
+                                                       maxWidth:self.maxWidth
+                                                      maxHeight:self.maxHeight
+                                            isMetadataAvailable:originalAsset != nil];
+            }
+            __block NSString *savedPath;
+            if (!originalAsset) {
+              // Image picked without an original asset (e.g. User pick image without permission)
+              savedPath =
+                  [FLTImagePickerPhotoAssetUtil saveImageWithPickerInfo:nil
+                                                                  image:localImage
+                                                           imageQuality:self.desiredImageQuality];
+              self.pathList[self.index] = savedPath;
+              [self completeOperation];
+            } else {
+              [[PHImageManager defaultManager]
+                  requestImageDataForAsset:originalAsset
+                                   options:nil
+                             resultHandler:^(
+                                 NSData *_Nullable imageData, NSString *_Nullable dataUTI,
+                                 UIImageOrientation orientation, NSDictionary *_Nullable info) {
+                               // maxWidth and maxHeight are used only for GIF images.
+                               savedPath = [FLTImagePickerPhotoAssetUtil
+                                   saveImageWithOriginalImageData:imageData
+                                                            image:localImage
+                                                         maxWidth:self.maxWidth
+                                                        maxHeight:self.maxHeight
+                                                     imageQuality:self.desiredImageQuality];
+                               self.pathList[self.index] = savedPath;
+                               [self completeOperation];
+                             }];
+            }
+          }
+        }];
+  }
```

Comment:
```
else {
   [self completeOperation];
}
```
for iOS 13 and below?

Reply:
We will never use this class with the lower versions of iOS 14. But still, it would be good to have something like this I guess :
```
else {
    [self setFinished:YES];
}
```